applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_hydraulic_solver.py

```diff
@@ -0,0 +1,692 @@
+        current_dt = self.main_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+        inlet_volume = -current_dt * inlet_discharge
+        outlet_volume = current_dt * outlet_discharge
+
+        # System water volume is calculated for current time step considering inlet and outlet discharge.
+        system_volume = inlet_volume + self.__initial_system_volume - outlet_volume
+        self.__initial_system_volume = system_volume
+
+
+
+    def _ComputeVolumeError(self):
+        if self.mass_source:
+            water_volume_after_transport = KratosCFD.FluidAuxiliaryUtilities.CalculateFluidNegativeVolume(self.GetComputingModelPart())
+
+            volume_error = (water_volume_after_transport -  self.__initial_system_volume) /  self.__initial_system_volume
+        else:
+            volume_error=0
+
+        self.main_model_part.ProcessInfo.SetValue(KratosCFD.VOLUME_ERROR, volume_error)
+
+
+
+    def __CalculateTimeIntegrationAcceleration(self):
+        # Acceleration for generalised alpha time integration method.
+        alpha_m=0.5*((3-self.rho_inf)/(1+self.rho_inf))
+        alpha_f=1/(1+self.rho_inf)
+        gamma= 0.5+alpha_m-alpha_f
+        dt=self.main_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+        for node in self.GetComputingModelPart().Nodes:
+            vn=node.GetSolutionStepValue(KratosMultiphysics.VELOCITY,1)
+            v = node.GetSolutionStepValue(
+                KratosMultiphysics.VELOCITY, 0)
+            acceleration_n=node.GetValue(KratosMultiphysics.ACCELERATION)
+            acceleration_alpha_method=(v-vn)/(gamma*dt)+((gamma-1)/gamma)*acceleration_n
+            node.SetValue(KratosMultiphysics.ACCELERATION,acceleration_alpha_method)
+
+    def __CalculateArtificialViscosity(self):
+
+        properties_1 = self.main_model_part.Properties[1]
+        water_dynamic_viscosity_max = self.artificial_limiter_coefficient * properties_1.GetValue(KratosMultiphysics.DYNAMIC_VISCOSITY)
+        for element in self.GetComputingModelPart().Elements:
+            artificial_viscosity_gauss = element.CalculateOnIntegrationPoints(KratosCFD.ARTIFICIAL_DYNAMIC_VISCOSITY, self.main_model_part.ProcessInfo)
+            item = 0
+            elem_artificial_viscosity= 0.0
+            for item in artificial_viscosity_gauss:
+                elem_artificial_viscosity += item
+                item +=1
+            elem_artificial_viscosity /= item
+            if elem_artificial_viscosity > water_dynamic_viscosity_max:
+                elem_artificial_viscosity = water_dynamic_viscosity_max
+            element.SetValue(KratosCFD.ARTIFICIAL_DYNAMIC_VISCOSITY, elem_artificial_viscosity)
+
+    def _PerformLevelSetConvection(self):
+        # Solve the levelset convection problem
+        self._GetLevelSetConvectionProcess().Execute()
+
+    def _PerformEulerianFmAleVelocity(self):
```

Comment:
```suggestion
    def __PerformEulerianFmAleVelocity(self):
```
I'd mangle this one.